### PR TITLE
Cmis write

### PR DESCRIPTION
**Pull request by bwrsandman**
Port of https://code.launchpad.net/~savoirfairelinux-openerp/knowledge-addons/cmis_write/+merge/213940 by @ehdem 

re-port of https://github.com/OCA/knowledge/pull/8

**Depends on #2**

Add cmis_write: It allows to create docs from OE in DMS
## LP reviews
- @max3903: Approve on 2014-04-19
- @bwrsandman: Needs Fixing on 2014-05-06
  
  > No unittests

### CODE REVIEW

**Comment by max3903** [comment]
:-1: 

I get this error message in the failing job to write a document in the DMS :

Traceback (most recent call last): 
File "/home/max/openerp/cmis/git/connector/connector/queue/worker.py", line 122, in run_job job.perform(session) 
File "/home/max/openerp/cmis/git/connector/connector/queue/job.py", line 491, in perform self.result = self.func(session, _self.args, *_self.kwargs) 
File "/home/max/openerp/cmis/git/cmis/cmis_write/ir_attachment.py", line 182, in create_doc_in_edm context=session.context) 
TypeError: _auth() takes at least 4 arguments (4 given)

**Comment by bwrsandman** [comment]
@max3903 simply pass `ids` when calling `_auth`

**Comment by ehdem** [comment]
Dependency with cmis module missing. I have to add the cmis module.

**Comment by bwrsandman** [comment]
@ehdem this is your pull request, by the way...

File: cmis_write/ir_attachment.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        for backend in cmis_backend_obj.browse(cr, uid, backend_ids,
+                                               context=context):
+            try:
+                repo = cmis_backend_obj._auth(cr, uid, backend.id,
+                                              context=context)
+                cmis_backend_rec = self.read(
+                    cr, uid, ids, ['id_dms'], context=context)[0]
+                id_dms = cmis_backend_rec['id_dms']
+                # Get results from id of document
+                results = repo.query(" SELECT * FROM  cmis:document WHERE \
```
Comment (JordiBForgeFlow) [inline cmis_write/ir_attachment.py]: I have tested this, and retrieves an empty list. What works instead of the query is:
result = repo.getObject(id_dms)

**Comment by JordiBForgeFlow** [comment]
I would like to help merge this PR, but I don't know how to do it! I have download cmis_write from the launchpad repository https://code.launchpad.net/~savoirfairelinux-openerp/knowledge-addons/cmis_write, but looks like that is a slightly different version.

**Comment by ehdem** [comment]
@jbeficent : Don't use  cmis_write from the launchpad repository https://code.launchpad.net/~savoirfairelinux-openerp/knowledge-addons/cmis_write. It's an old version

**Comment by ehdem** [comment]
The cmis module is already in https://github.com/OCA/connector-cmis

**Comment by max3903** [comment]
:+1:

**Comment by max3903** [comment]
Visual check on runbot.

:+1:

File: cmis_write/ir_attachment.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+        ),
+    }
+
+
+@job
+def create_doc_in_edm(session, model_name, value, res,
+                      dict_metadata, user_login, filters=None):
+    """
+    This method allows to create a doc from Odoo to DMS
+    """
```
Comment (ghost) [inline cmis_write/ir_attachment.py]: @ehdem You don't seem to use any connector unit. Why? Connector units are the core of the connector framework. Without using these, your code will be hard to maintain and inherit. Also, you don't make any use of the framework's exceptions. How will the worker know whether to retry a job or to cancel it?

**Comment by bwrsandman** [comment]
:-1: That is my issue with the connector too.

This doesn't seem to properly use the connector framework.

File: cmis_write/ir_attachment.py (hunk 2 above)
Comment (ehdem) [inline cmis_write/ir_attachment.py]: @dufresnedavid : Yes, I don't use it yet. http://odoo-connector.com/guides/concepts.html#connectorunit

**Comment by lmignon** [comment]
With the commit  d469e46, this PR depends of https://github.com/OCA/connector-cmis/pull/7
For you  information, the code also works on 8.0 as it is.

**Pull request merged**